pcdsdevices/ccm.py

```diff
@@ -773,6 +774,198 @@ def inverse(self, real_pos: namedtuple) -> namedtuple:
+            finishes.
+        stepsize: float
+            Energy increment after a failed check on the IPM ratio. Should be smaller
+            than the self-seed bandwidth for good results.
+        max_steps: int
+            If no good point is reached after max_step points, the optimization is
+            aborted.
+        """
+        self.self_seed.move(setpoint)
+        # move slightly below the setpoint
+        status = super().move(setpoint-offset, wait=True)
+        # wait for self-seed to settle
+        time.sleep(1)
+
+        for step in range(max_steps):
+            intensities = self.get_ipms_intensities(n_average)
+            ipm_ratio = self.ipm_ratio(intensities)
+            if ipm_ratio > self.ipm_ratio_threshold:
+                print('Optimization complete!\n')
+                break
+            status = super().mvr(stepsize, wait=True)
+        return status
+
+    def get_ipms_intensities(self, n_shots=10):
+        intensities = []
+        status = Status(timeout=5)
+
+        def cb(*args, obj: OphydObject, sub_type: str, **kwargs) -> None:
+            intensities.append(obj.value)
+            if len(intensities) >= n_shots:
+                status.set_finished()
+
+        cid = self.ipms.subscribe(cb)
+        status.wait(5)
+        self.ipms.unsubscribe(cid)
+
+        # for ii in range(nn):
+        #     intensities.append(self.ipms.get())
+        #     time.sleep(sleep)
+        return np.asarray(intensities)
+
+    def ipm_ratio(self, intensities):
+        """ Calculate the mean(downstream)/mean(upstream) IPMs """
+        ratio = np.average(intensities, axis=0)
+        ratio = ratio[self.ipm_idx['downstream']] / ratio[self.ipm_idx['upstream']]
+        return ratio
+
+    def setup_method_monotonic(self):
+        """ Helps the user throught the setup of the parameters for the monotonic
+        search algorithm. This should be handled by someone who knows what they
+        are doing.
+        """
+        print('Setting up monotonic search parameters.')
+
+        s = 'Offset (eV) (default: TBD): '
+        offset = float(input(s) or 5)
+
+        s = 'Number of shots for the IPMs average (default: 50): '
+        n = int(input(s) or 50)
+
+        s = 'IPM ratio threshold (default: TBD): '
+        ratio_th = float(input(s) or 2.5)
+
+        s = 'Energy increment (eV) (default: TBD): '
+        step_size = float(input(s) or 0.5)
+
+        s = 'Maximum number of optimization steps (default: 10): '
+        max_step = int(input(s) or 10)
+
+        self.optimization_args['monotonic'] = {
+            'energy_offset' : offset,
+            'n_average' : n,
+            'ipm_ratio_threshold' : ratio_th,
+            'energy_step_size' : step_size,
+            'max_step' : max_step
+        }
+        self.method = 'monotonic'
+        return
+
+    def move(self, new_pos , *args, **kwargs):
+        status = self.method_map[self.method](new_pos, self.optimization_args[self.method])
```

Comment:
```suggestion
        status = self.method_map[self.method](new_pos, **self.optimization_args[self.method])
```

You'll have to unpack the optimization kwargs here, but otherwise I think this should work